sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluate/_evaluate.py

```diff
@@ -60,6 +60,12 @@
 
 LOGGER = logging.getLogger(__name__)
 
+# Tag validation constants
+MAX_TAG_COUNT = 50
+MAX_TAG_COUNT_WITHOUT_DEFAULT = 49
+MAX_TAG_KEY_LENGTH = 100
+MAX_TAG_VALUE_LENGTH = 1000
```

Comment:
There are system tags that are added by service which would account for this limit. So would be good to just surface the docs as part of error message https://learn.microsoft.com/en-us/azure/machine-learning/resource-limits-capacity?view=azureml-api-2#runs